scripts/stylable-metadata-plugin.js

```diff
@@ -0,0 +1,190 @@
+    ([testName]) => testName === `Default${name}`
+  )
+
+  if (!defaultVarEntry) {
+    return ''
+  }
+
+  const [,info] = defaultVarEntry
+
+  return info.value
+}
+
+const guessVarType = (name, value, defaultValue) => {
+  const colorRegexp = /color/i
+
+  const colorRules = [
+    () => name.match(colorRegexp),
+    () => value.match(colorRegexp),
+    () => defaultValue.match(colorRegexp)
+  ]
+
+  const fontRegexp = /font/i
+  const fontGuessRegexp = /[^A-Za-z_-]theme\:/
+  
+  const fontRules = [
+    () => name.match(fontRegexp),
+    () => value.match(fontRegexp),
+    () => defaultValue.match(fontRegexp),
+    () => defaultValue.match(fontGuessRegexp),
+  ]
+
+  const succeedingRule = rule => Boolean(rule())
+
+  if (colorRules.find(succeedingRule)) {
+    return 'color';
+  } else if (fontRules.find(succeedingRule)) {
+    return 'font'
+  } else {
+    return 'number'
+  }
+}
+
+const parseComment = comment => {
+  if (!comment) {
+    return {
+      description: '',
+    }
+  }
+
+  const supportedTags = ['type', 'default', 'min', 'max', 'unit']
+  const commentParts = comment.split('@').map(part => part.trim())
+
+  const description = commentParts.shift()
+
+  const result = {description}
+
+  commentParts.forEach(part => {
+    for (const supportedTag of supportedTags) {
+      if (part.startsWith(`${supportedTag} `)) {
+        result[supportedTag] = part.substring(supportedTag.length + 1)
+      }
+    }
+  })
+
+  return result
+}
+
+const getOverridableVars = ast => {
+  const allVars = getAllStylableVars(ast)
+
+  const vars = Object.entries(allVars)
+    .reduce((result, [name, info]) => {
+      if (info.value.startsWith('-')) {
+        const defaultValue = getDefaultValue(name, allVars)
+        const commentInfo = parseComment(info.comment)
+
+        result.push({
+          name,
+          type: commentInfo.type || guessVarType(name, info.value, defaultValue),
+          defaultValue: commentInfo.default || defaultValue,
+          description: commentInfo.description
+        })
+      }
+
+      return result
+    }, [])
+
+  return vars
+}
+
+module.exports = async function ({source, metadata, basePath}) {
+  const relativeComponentPath = await pathFinder(source)
+  const absoluteComponentPath = path.resolve(basePath, relativeComponentPath)
+
+  const componentSource = fs.readFileSync(absoluteComponentPath, {encoding: 'utf8'})
+  const componentAst = parse(componentSource)
+
+  const relativeStylablePath = await getImportedStylablePath(componentAst)
+
+  if (!relativeStylablePath) {
+    return {metadata}
+  }
+
+  const absoluteStylablePath = path.resolve(path.dirname(absoluteComponentPath), relativeStylablePath)
+
+  const stylableSource = fs.readFileSync(absoluteStylablePath, {encoding: 'utf8'})
+  const stylableAst = parseStylable(stylableSource)
+
+  return {
+    metadata: {
+      ...metadata,
+      stylable: {
+        overridableVars: getOverridableVars(stylableAst)
+      }
+    }
+  }
+}
```

Comment:
Let's move this file to the `.storybook` folder, for now.
We usually use the `scripts` folder for scripts that we run on CI, or build, etc.
And since this file is only used in the `.storybook` webpack configuration, I think we can place it there.